test/units/modules/test_iptables.py

```diff
@@ -1,1174 +1,1402 @@
+    ]
+    assert second_cmd_args_list[0][0] == [
+        IPTABLES_CMD,
+        "-t",
+        "nat",
+        "-D",
+        "PREROUTING",
+        "-p",
+        "udp",
+        "-s",
+        "1.2.3.4/32",
+        "-d",
+        "7.8.9.10/42",
+        "-j",
+        "SNAT",
+        "--to-source",
+        "5.5.5.5/32",
+        "-i",
+        "eth0",
+        "-o",
+        "eth1",
+        "--source-port",
+        "22",
+        "--to-ports",
+        "8600",
+        "-m",
+        "comment",
+        "--comment",
+        "this is a comment",
+    ]
+
+
+def test_remove_rule_check_mode(mocker, mock_basic_commands):
+    """Test flush without parameters check mode."""
+    set_module_args(
+        {
+            "chain": "PREROUTING",
+            "source": "1.2.3.4/32",
+            "destination": "7.8.9.10/42",
+            "jump": "SNAT",
+            "table": "nat",
+            "to_source": "5.5.5.5/32",
+            "protocol": "udp",
+            "source_port": "22",
+            "to_ports": "8600",
+            "state": "absent",
+            "in_interface": "eth0",
+            "out_interface": "eth1",
+            "comment": "this is a comment",
+            "_ansible_check_mode": True,
+        }
+    )
+
+    commands_results = [
+        (0, "", ""),
+    ]
+
+    run_command = mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.run_command",
+        side_effect=commands_results,
+    )
+
+    with pytest.raises(SystemExit):
+        iptables.main()
+
+    assert run_command.call_count == 1
+    first_cmd_args_list = run_command.call_args_list[0]
+    assert first_cmd_args_list[0][0] == [
+        IPTABLES_CMD,
+        "-t",
+        "nat",
+        "-C",
+        "PREROUTING",
+        "-p",
+        "udp",
+        "-s",
+        "1.2.3.4/32",
+        "-d",
+        "7.8.9.10/42",
+        "-j",
+        "SNAT",
+        "--to-source",
+        "5.5.5.5/32",
+        "-i",
+        "eth0",
+        "-o",
+        "eth1",
+        "--source-port",
+        "22",
+        "--to-ports",
+        "8600",
+        "-m",
+        "comment",
+        "--comment",
+        "this is a comment",
+    ]
+
+
+@pytest.mark.parametrize(
+    ("test_input", "expected"),
+    [
+        pytest.param(
             {
-                'chain': 'OUTPUT',
-                'protocol': 'tcp',
-                'jump': 'DROP',
-                'tcp_flags': 'flags=ALL flags_set="ACK,RST,SYN,FIN"'
+                "chain": "INPUT",
+                "protocol": "tcp",
+                "reject_with": "tcp-reset",
+                "ip_version": "ipv4",
             },
+            [
+                IPTABLES_CMD,
+                "-t",
+                "filter",
+                "-C",
+                "INPUT",
+                "-p",
+                "tcp",
+                "-j",
+                "REJECT",
+                "--reject-with",
+                "tcp-reset",
+            ],
+            id="insert_reject_with",
```

Comment:
nit: dash-separated `kebab-style` identifiers usually look nicer in pytest test reports. And they are easier to spot as separate from module/function identifiers on the same line that are limited to `snake_case`.